semanticdb/semanticdb3/semanticdb3.md

```diff
@@ -2244,11 +2241,839 @@ occurrences. We intend to improve on this in the future.
+  </tr>
+  <tr>
+    <td><code>PENNY</code></td>
+    <td><code>a.Coin#PENNY.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;Coin&gt;, List())</code></td>
+  </tr>
+  <tr>
+    <td><code>NICKEL</code></td>
+    <td><code>a.Coin#NICKEL.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;Coin&gt;, List())</code></td>
+  </tr>
+  <tr>
+    <td></td>
+    <td><code>a.Coin#values().</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;Array&gt;, List(&lt;Coin&gt;)))</code></td>
+  </tr>
+  <tr>
+    <td></td>
+    <td><code>a.Coin#valueOf(String).</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;Coin&gt;, List()))</code></td>
+  </tr>
+</table>
+
+Notes:
+* Enum declarations follow the same rules as class declarations.
+* Supported properties for enum declarations are:
+  * `FINAL`: implicitly set for all enum declarations.
+  * `STATIC`: implicitly set for all enum declarations.
+  * `ENUM`: implicitly set for all enum declarations.
+* JLS mandates the following synthetic members for enum declarations
+  [\[86\]][86]:
+  * Enum fields have kind `FIELD`, properties `FINAL`, `STATIC` and `ENUM`,
+    are named after the corresponding enum constants, have the type of the
+    enum declaration and `PUBLIC` accessibility.
+  * `valueOf` has kind `METHOD`, property `STATIC`, have a method type that
+    goes from a `<String>` parameter to the enum declaration and
+    `PUBLIC` accessibility.
+  * `values` has kind `METHOD`, property `STATIC`, have a method type that
+    goes from an empty parameter list to an array of the enum declaration
+    and `PUBLIC` accessibility.
+* Enum declarations support [all Java accessibilities](#java-accessibility).
+
+**Interface declarations** [\[77\]][77] are represented by a single symbol
+like classes but with the `INTERFACE` kind.
+
+```java
+package a;
+
+public interface List<T> extends I {
+  T head();
+}
+```
+
+<table>
+  <tr>
+    <td><b>Definition</b></td>
+    <td width="200px"><b>Symbol</b></td>
+    <td><b>Kind</b></td>
+    <td><b>Signature</b></td>
+  </tr>
+  <tr>
+    <td><code>List</code></td>
+    <td><code>a.List#</code></td>
+    <td><code>CLASS</code></td>
+    <td><code>ClassInfoType(List(&lt;T&gt;), List(&lt;I&gt;), List(&lt;head&gt;))</code></td>
+  </tr>
+  <tr>
+    <td><code>head</code></td>
+    <td><code>a.List#head().</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;T&gt;, List()))</code></td>
+  </tr>
+</table>
+
+The differences between interface symbols and class symbols are:
+
+* Interfaces do not have constructors.
+* Supported properties for interface symbols are:
+  * `ABSTRACT`: implicitly set for all interface symbols.
+* Interface declarations support
+  [all Java accessibilities](#java-accessibility).
```

Comment:
Field declarations in interfaces can only be public and `STATIC` `FINAL`
https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.3

Method declarations in interfaces can only be public
https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.4

Not sure if this should be mentioned in interfaces of in methods/fields.

Reply:
Done.